File: client/bindings/nodejs/examples/consolidation.ts
// Copyright 2021-2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, CoinType, initLogger } from '@iota/client';
require('dotenv').config({ path: '../.env' });

// Run with command:
// node ./dist/consolidate.js

// In this example we will consolidate all funds in a range of addresses
async function run() {
    initLogger();
    if (!process.env.NODE_URL) {
        throw new Error('.env NODE_URL is undefined, see .env.example');
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL],
        localPow: true,
    });

    try {
        if (!process.env.NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1) {
            throw new Error('.env mnemonic is undefined, see .env.example');
        }

        // Configure your own mnemonic in ".env". Since the output amount cannot be zero, the mnemonic must contain non-zero
        // balance
        const secretManager = {
            mnemonic: process.env.NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1,
        };

        // Here all funds will be sent to the address with the lowest index in the range
        const address = await client.consolidateFunds(secretManager, {
            coinType: CoinType.Shimmer,
            accountIndex: 0,
            range: {
                start: 0,
                end: 10,
            },
            internal: false,
        });
        console.log('Funds consolidated to: ', address);
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());

File: client/examples/high_level/consolidation.rs
// Copyright 2021 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

//! cargo run --example consolidation --release

use iota_client::{
    api::GetAddressesBuilderOptions,
    secret::{mnemonic::MnemonicSecretManager, SecretManager},
    Client, Result,
};

/// In this example we will consolidate all funds in a range of addresses

#[tokio::main]
async fn main() -> Result<()> {
    // This example uses dotenv, which is not safe for use in production
    // Configure your own mnemonic in ".env". Since the output amount cannot be zero, the mnemonic must contain non-zero
    // balance
    dotenv::dotenv().ok();

    let node_url = std::env::var("NODE_URL").unwrap();

    let address_range = 0u32..150;
    // Create a client instance
    let client = Client::builder().with_node(&node_url)?.finish()?;

    let secret_manager =
        SecretManager::try_from_hex_seed(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_SEED_1").unwrap())?;

    // Here all funds will be send to the address with the lowest index in the range
    let address = client
        .consolidate_funds(
            &secret_manager,
            GetAddressesBuilderOptions {
                range: Some(address_range),
                ..Default::default()
            },
        )
        .await?;

    println!("Funds consolidated to {}", address);
    Ok(())
}
